libobs/util/platform-nix.c

```diff
@@ -96,10 +102,56 @@ void os_dlclose(void *module)
 		dlclose(module);
 }
 
+#if !defined(__APPLE__) && OBS_QT_VERSION == 6
+int module_has_qt5_check(const char *path)
+{
+	void *mod = os_dlopen(path);
+	if (mod == NULL) {
+		return 1;
+	}
+
+	struct link_map *list = NULL;
+	if (dlinfo(mod, RTLD_DI_LINKMAP, &list) == 0) {
+		for (struct link_map *ptr = list; ptr; ptr = ptr->l_next) {
+			char *slash = strrchr(ptr->l_name, '/');
+			if (!slash)
+				continue;
+			char *base = slash + 1;
+			if (strncmp(base, "libQt5", 6) == 0) {
+				return 0;
+			}
+		}
+	}
+
+	return 1;
+}
+
+void fork_and_qt5_check(const char *path, bool *can_load)
+{
+	pid_t pid = fork();
+	if (pid == 0) {
+		_exit(module_has_qt5_check(path));
+	}
+	if (pid < 0) {
+		return;
+	}
+	int status;
+	if (waitpid(pid, &status, 0) < 0) {
+		return;
+	}
+	if (WIFEXITED(status) && WEXITSTATUS(status) == 0) {
+		*can_load = false;
+	}
```

Comment:
Minor nit, allow me to suggest the following style:

```c
void fork_and_check_qt5(const char *path, bool *can_load)
{
	pid_t pid = fork();


	if (pid == -1) {
		blog(LOG_WARNING, "Error with fork() call");
		*can_load = false;
		return;
	}

	if (pid == 0) {
		// This is the child process
		_exit(module_has_qt5_check(path));
	} else {
		int status;
		if (waitpid(pid, &status, 0) >= 0) {
			*can_load = WIFEXITED(status) && WEXITSTATUS(status) != 0;
		} else {
			blog(LOG_WARNING, "Error waiting for child process");
			*can_load = false;
		}
	}
}
```

Reply:
this introduces more possible failures in loading plugins, which I am specifically trying to avoid. We should only avoid loading these plugins, which causes the incompatible prompt to be shown, if we are quite sure they are incompatible.  Otherwise let OBS behave exactly as it did before.

Reply:
Then `s/can_load = false/can_load = true/g` :slightly_smiling_face: 

I just enjoy the quality of being explicit in what it returns

Reply:
can_load is already prepared, we are only setting it to false if we check and find qt5. Its not being returned so to speak. We could return a value if you want.

Personally i find this branching construction more confusing. Previously we have straightline code for the parent with branches for the child and failures, instead this keeps a growing chain of else's.

Reply:
Updated with return, should appear simpler now.